aodupechildimport.php

```diff
@@ -157,6 +157,46 @@ function aodupechildimport_civicrm_pre($op, $objectName, $id, &$params) {
   }
 }
 
+  function aodupechildimport_civicrm_post($op, $objectName, $objectId, &$objectRef) {
+    if ($objectName == "Relationship" && $op == 'create') {
+      // This is done at the time of creating a new relationship between the child and parent.
+      $childRel = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_RelationshipType', 'Child of', 'id', 'name_a_b');
+      if ($objectRef->relationship_type_id == $childRel) {
+        createSharedAddress($objectRef->contact_id_b, $objectRef->contact_id_a);
+      }
+    }
+    if ($objectName == 'Individual' && $op == 'edit'
+      && !empty($objectRef->contact_sub_type) && strpos($objectRef->contact_sub_type, 'Child') !== false) {
+      // This is done if the child exists, and is an update. The method above isn't called for existing relationships.
+      $childRel = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_RelationshipType', 'Child of', 'id', 'name_a_b');
+      $relationships = civicrm_api3('Relationship', 'get', ['contact_id_a' => $objectId, 'relationship_type_id' => $childRel]);
+      if (!empty($relationships['values'])) {
+        foreach ($relationships['values'] as $relationship) {
+          deleteChildAddresses($relationship['contact_id_a']);
+          createSharedAddress($relationship['contact_id_b'], $relationship['contact_id_a']);
+        }
+      }
+    }
+  }
+
+  function deleteChildAddresses($childId) {
```

Comment:
OK lets think about this:

The places that might trigger an  address creation or edit are likely what:

* - Parent Child registration
* - Imports
* - ? 

So this expands it across to include when a backoffice user creates a relationship which should be ok. In regards to My Address code I don't believe this will have negative impacts on this

So what would happen if you had a different address for the different parents? (thinking separated couple) This would potentially destroy a good address right?. Also is there a reason why the import wouldn't just update the current address of the parent and propogate that onto the Child or does the import not work like that?